src/dune_engine/build_system.ml

```diff
@@ -2066,7 +2087,8 @@ end = struct
         , dir
         , alias
         , capture_stdout
-        , can_go_in_shared_cache )
+        , can_go_in_shared_cache
```

Comment:
This line in the diff reminds me that we should probably force `can_go_in_shared_cache` to be `false` when `patch_back_source_tree = true`. Adding support for the shared cache for this mode seems difficult.

Reply:
Indeed, I adapted the code to set `can_go_in_shared_cache = false` for actions that also have `patch_back_source_tree = true`